com.unity.netcode.gameobjects/CHANGELOG.md

```diff
@@ -27,6 +27,7 @@ Additional documentation and release notes are available at [Multiplayer Documen
 - Fixed NetworkList to properly call INetworkSerializable's NetworkSerialize() method (#1682)
 - Fixed The NetworkConfig's checksum hash includes the NetworkTick so that clients with a different tickrate than the server are identified and not allowed to connect (#1728)
 - Fixed OwnedObjects not being properly modified when using ChangeOwnership (#1731)
+- Fixed an error where the snapshot system was spawning all objects as scene objects, which caused objects that were spawned during a client's session to not despawn when the client disconnected, and thus not respawn properly when they reconnected, leading to NetworkVariable updates for that object being lost. (#1740)
```

Comment:
I'm a little bit confused. Don't we have SnapshotSystem disabled by default?

Reply:
I believe what happened is that this bug was filed while `experimental-develop` was `develop` and snapshot was active by default. The fix is good and would be needed on that branch. But not necessarily on the current `develop`.

The problem I now see is that once we commit this fix to the new `develop`, `experimental-develop` becomes a bit more divergent and will be harder to merge back in. So, we could consider also merging this PR into `experimental-develop`

Reply:
Yeah, I didn't realize we'd completely disabled snapshot spawns. I thought they were still in use. I'll retarget this to experimental-develop.